cpp/lib/src/MapUtil.cc

```diff
@@ -86,9 +86,10 @@ void DeserialiseMap(const std::string &input_filename, std::unordered_map<std::s
         if (key.empty() or value.empty())
             LOG_ERROR("Bad input in \"" + input_filename + "\" on line " + std::to_string(line_no) + "!");
         if (revert_keys_and_values)
-            (*map)[value] = key;
-        else
-            (*map)[key] = value;
+            key.swap(value);
+        if (unlikely(map->find(key) != map->end()))
```

Comment:
There might be a problem when revert_keys_and_values is set and multiple different keys would have the same values, so we might want to check this before swapping as well